Complete Java coding rules (from internal documentation). (#299)
MicroEJ/docs: Tutorials/tutorialImproveCodeQuality.rst

  parameter is renamed afterward, the comment is not changed
  automatically. Prefer using *the given xxx*.

Code Convention
^^^^^^^^^^^^^^^

Official documentation:
https://www.oracle.com/java/technologies/javase/codeconventions-introduction.html

Class Declaration
~~~~~~~~~~~~~~~~~

The parts of a class or interface declaration must appear in the order suggested by the Code Convention for the Java Programming Language:

- Constants.
- Class (static) fields.
- Instance fields.
- Constructors
- Methods

Fields Order
~~~~~~~~~~~~

For a better readability, the fields (class and instance) must be ordered by scope:

> 1. ``public``
> 2. ``protected``
> 3. ``package``
> 4. ``private``

Reviewer: ```suggestion
#. ``public``
#. ``protected``
#. ``package``
#. ``private``
```

Methods Order
~~~~~~~~~~~~~

It is recommended to group related methods together.
It helps for maintenance:

> - to search a bug on a specific feature,
> - to split a class into several ones.

Reviewer: ```suggestion
- when searching for a bug on a specific feature,
- when refactoring a class into several ones.
```
